cli/emit.rs

```diff
@@ -452,19 +475,25 @@ pub(crate) fn check_and_maybe_emit(
           log::debug!("skipping emit for {}", specifier);
           continue;
         }
+        let mut emit_data_item = emit_data_items
+          .entry(specifier.clone())
+          .or_insert_with(|| SpecifierEmitData {
+            version_hash: get_version(source_bytes, &config_bytes),
+            text: None,
+            declaration: None,
+            map: None,
+          });
         match emit.media_type {
           MediaType::JavaScript | MediaType::Mjs | MediaType::Cjs => {
-            let version = get_version(source_bytes, &config_bytes);
-            cache.set(CacheType::Version, &specifier, version)?;
-            cache.set(CacheType::Emit, &specifier, emit.data)?;
```

Comment:
An example of a previous bug would be if the process got killed right before entering this function call.